db/migrate/20180816134832_cast_lookup_key_values.rb

```diff
@@ -0,0 +1,37 @@
+class CastLookupKeyValues < ActiveRecord::Migration[5.1]
+  def up
+    # Different LookupKey types handle casting a bit differently
+    PuppetclassLookupKey.unscoped.preload(:lookup_values).where(override: true).where.not(key_type: 'string').find_each do |key|
+      cast_key_and_values(key)
+    end
+
+    VariableLookupKey.unscoped.preload(:lookup_values).where.not(key_type: 'string').find_each do |key|
+      cast_key_and_values(key)
+    end
+  end
+
+  private
+
+  def cast_key_and_values(key)
+    fix_value(key, :default_value)
+    key.lookup_values.each do |lv|
+      fix_value(lv, :value)
+    end
+  end
+
+  def safemode
+    @box ||= Safemode::Box.new
+  end
+
+  def fix_value(obj, attribute)
+    return if obj.omit
```

Comment:
I confirm removing this line solves my issue of having sting values on omitted default params; ultimately defeating ENC by calling deep_merge on a string (`NoMethodError: undefined method 'deep_merge!' for "{}":String`).

Reply:
@tbrisker should we remove that line?

Reply:
opened https://github.com/theforeman/foreman/pull/5995 to fix this case and improve error message.